static/src/javascripts/projects/common/modules/ui/first-pv-consent-banner.js

```diff
@@ -61,10 +62,67 @@ const bindableClassNames: BindableClassNames = {
     agree: 'js-first-pv-consent-agree',
 };
 
+let inModalTestRegularVariant;
+const isInModalTestRegularVariant = (): boolean => {
```

Comment:
I suggest instead:
```
const isInModalTestRegularVariant = (): boolean => {
    if (inModalTestRegularVariant === undefined) {
        inModalTestRegularVariant = isInVariantSynchronous(
            commercialConsentModalBanner,
            'regularVariant'
        );
    }

    return inModalTestRegularVariant;
};
```

and similar for the functions below.

Reply:
Minor: do we need `inModalTestRegularVariant` to be declared outside the function? Also I find `let` a bit harder to reason about than `const` since it can be reassigned (hence preferring const) 😅

```js
const isInModalTestRegularVariant = (): boolean => {
    const inModalTestRegularVariant = isInVariantSynchronous(
            commercialConsentModalBanner,
            'regularVariant'
        );
    return inModalTestRegularVariant;
};
```

or even:

```js
const isInModalTestRegularVariant: boolean = isInVariantSynchronous(commercialConsentModalBanner, 'regularVariant');
```


Would this work?

Reply:
@katebee my first implementation actually was exactly as you suggested, you can see where I removed it here: https://github.com/guardian/frontend/pull/21464/commits/fa56dec3ed9c863b42ebc0b20ea2d9cf210cdd29 

The reason I removed this was because it was causing issues when testing as a side side effect of simply loading `first-pv-consent-banner.js` in `first-pv-consent-banner.spec.js` was that these variables were being set before we have any opportunity to mock `isInVariantSynchronous` and force their value to what we'd like.

I opted to use `let`s such as `let inModalTestRegularVariant` because I need to run these checks in multiple places within the scope of this module and I didn't think it was efficient to call `isInVariantSynchronous(commercialConsentModalBanner, 'regularVariant')` multiple times (when the result will remain the same every time), so I opted to call `isInVariantSynchronous(commercialConsentModalBanner, 'regularVariant')` once and then assign the result to the `let` `inModalTestRegularVariant` which we check against on every call of `isInModalTestRegularVariant`.